libraries/core_libs/network/src/tarcap/packets_handlers/votes_dag_sync_packet_handler.cpp

```diff
@@ -0,0 +1,114 @@
+                 << ". The peer may be a malicIous player, will be disconnected";
+    disconnect(packet_data.from_node_id_, dev::p2p::UserReason);
+    return;
+  }
+
+  auto zero_vote = std::make_shared<Vote>(packet_data.rlp_[0].data().toBytes());
+
+  const auto current_pbft_round = pbft_mgr_->getPbftRound();
+  const auto votes_pbft_round = zero_vote->getRound();
+
+  // Dag sync votes are related to the previous finalized pbft period
+  // -> current_pbft_round must be greater than votes_pbft_round
+  if (votes_pbft_round >= current_pbft_round) {
+    LOG(log_wr_) << "Received votes_pbft_round(" << votes_pbft_round << ") >= current_pbft_round(" << current_pbft_round
+                 << ")";
+    return;
+  }
+
+  const auto current_pbft_period = pbft_chain_->getPbftChainSize();
+
+  // Process each vote
+  for (size_t i = 0; i < votes_count; i++) {
+    auto vote = std::make_shared<Vote>(packet_data.rlp_[i].data().toBytes());
+
+    // Check if all received votes have the same round
+    if (vote->getRound() != votes_pbft_round) {
+      LOG(log_er_) << "Received dag sync votes bundle with unmatched rounds from " << packet_data.from_node_id_
+                   << ". The peer may be a malicious player, will be disconnected";
+      disconnect(packet_data.from_node_id_, dev::p2p::UserReason);
+      return;
+    }
+
+    const auto vote_hash = vote->getHash();
+    LOG(log_dg_) << "Received dag sync vote " << vote_hash;
+    peer->markVoteAsKnown(vote_hash);
+
+    // TODO: check if current_pbft_period - 1 pbft_block hash == vote->blockHash
+
+    // TODO: Check if the voter address is unique. Same voter should be able to vote for the same block twice -> double
+    // rewards
+
+    // Check if the voter account is valid, malicious vote
+    addr_t voter = vote->getVoterAddr();
+    auto vote_weighted_index = vote->getWeightedIndex();
+    size_t voter_dpos_votes_count = 0;
+
+    // get voter dpos eligible vote count
+    try {
+      voter_dpos_votes_count = final_chain_->dpos_eligible_vote_count(current_pbft_period - 1, voter);
+    } catch (state_api::ErrFutureBlock &c) {
+      LOG(log_er_) << c.what() << ". Period " << current_pbft_period - 1
+                   << " is too far ahead of DPOS, have executed chain size " << final_chain_->last_block_number();
+      // TODO: should all these failed checks do continue or return ?
+      continue;
+    }
+
+    if (vote_weighted_index >= voter_dpos_votes_count) {
+      LOG(log_dg_) << "Account " << voter << " was not eligible to vote. Vote: " << vote;
+      continue;
+    }
+
+    // TODO: pbft_mgr_->getPreviousPbftPeriodDposTotalVotesCount and
+    // pbft_mgr_->getPreviousPbftPeriodSortitionThreshold() mght be incosistent with current_pbft_period - 1
+    if (auto result = vote->validate(pbft_mgr_->getPreviousPbftPeriodDposTotalVotesCount(),
+                                     pbft_mgr_->getPreviousPbftPeriodSortitionThreshold());
```

Comment:
You're right, getPreviousPbftPeriodDposTotalVotesCount and getPreviousPbftPeriodSortitionThreshold might be inconsistent.
I think one solution we could save all DPOS total votes count and sortition threshold for all periods in DB, not only save for previous period. And add period in vote structure. Then could use vote->getPeriod() to get DPOS total votes count and sortition threshold.